packages/read-model/src/Brand/BrandView.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\ReadModelBundle\Brand;
+
+/**
+ * Class representing brand on frontend
+ *
+ * @see https://docs.shopsys.com/en/latest/model/introduction-to-read-model/
+ */
+class BrandView
+{
+    /**
+     * @var int
+     */
+    protected $id;
+
+    /**
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * @var string
+     */
+    protected $mainUrl;
+
+    /**
+     * @param int $id
+     * @param string $name
+     * @param string $mainUrl
+     */
+    public function __construct(int $id, string $name, string $mainUrl)
+    {
+        $this->id = $id;
+        $this->name = $name;
+        $this->mainUrl = $mainUrl;
```

Comment:
I belive the BrandView can exists withou main URL. IMO it is too much information for it.

e.g. it is totally useless for you when serving data for (any) API unless it is required extra.

Reply:
hmm... interesting idea. I don't plan to use read model objects in API. 
In product detail, I need a brand URL, so I have two options here. 

- To not have URL in the object and generate it in the template itself (for me it's against the idea of read model and it can worsen the performance if used heavily)
- Have the second object for the brand on the product detail, then I wonder why to have this one without URL at all, when it will not be used.

Can you please explain more to me the possible solution you have?

Reply:
I will try to explain my point:

It may seem a bit as a YAGNI thing from me (oscilating with SRP) but I would not tighten these views with all the informations just because we use it that way here. I accept the poiny wit API. So just imagine the case we add any page/component where it is not used.

What I saw here and still keep in mind are only the URLs (you need to create them via FriendlyUrlFacade) and images (different images, different aproaches, many informations needed). These are the things which we really do not need everytime. Sure, we do need them now it current twigs. What happens when we add url/image less twig? Use read-model for feeds? Maybe even browsing website vie teletext (maybe that one is ridiculous)?.

Hope I summarize my points clear. What do you think? Do you see any chance of this kind of usage?

Reply:
I'm not sure if I understand properly. 

So you suggest not have a URL in the view object and get the URL with Twig extension in the template?